homeassistant/components/shelly/binary_sensor.py

```diff
@@ -19,6 +22,17 @@
     async_setup_entry_rest,
 )
 
+
+def is_momentary_input(settings: dict, block: aioshelly.Block) -> bool:
+    """Return true if input button settings is set to a momentary type."""
+    button = settings.get("relays") or settings.get("lights") or settings.get("inputs")
+
+    # In Shelly Dimmer (SHDM) the first channel set the button type for all channels
+    channel = min(int(block.channel or 0), len(button) - 1)
+
+    return button[channel]["btn_type"] in ["momentary", "momentary_on_release"]
```

Comment:
What's the rationale here? i3s also have "toggle" and that should work as input, no?

Reply:
This is per channel:
If channel in I3 is set to `toggle` or `edge` it will create a binary sensor for it.
 If it is set to `momentary` or `momentary_on_release`  an entity will not be created.
This is HA requirement, you can read about it here:
https://github.com/home-assistant/core/pull/42700#discussion_r517629879